frontend/src/context/RecipeContext.js

```diff
@@ -0,0 +1,40 @@
+import { createContext, useState, useEffect } from "react";
+
+export const RecipeContext = createContext();
+
+export const RecipeProvider = ({ children }) => {
+    const [selectedRecipes, setSelectedRecipes] = useState('');
+  const [recipes, setRecipes] = useState([]);
+
+  const getRecipes = async () => {
+    try {
+      const response = await fetch("http://localhost:3100/api/recipes");
+      const jsonData = await response.json();
+      setRecipes(jsonData);
+    } catch (error) {
+      console.error(error.message);
+    }
+  };
+
+  useEffect(() => {
+    getRecipes();
+  }, []);
+
+
+  const getRandomRecipes = () => {
+    const randomRecipes = new Set();
+    while (randomRecipes.size < 5) {
```

Comment:
This will hang, if there's less than five recipes in the database. Could 5 be a named constant or a function argument?

```suggestion
    while (randomRecipes.size < Math.min(NUMBER_OF_RANDOM_RECIPES, recipes.length)) {
```